src/zenml/config/schedule.py

```diff
@@ -43,17 +52,44 @@ class Schedule(BaseModel):
             schedules the latest interval if more than one interval is ready to
             be scheduled. Usually, if your pipeline handles backfill
             internally, you should turn catchup off to avoid duplicate backfill.
-        run_once_start_time: datetime object to indicate when to run the
-            pipeline once. This is useful for one-off runs.
+        run_once_start_time: When to run the pipeline once. If this is a
+            datetime object without any timezone, it is treated as a datetime
+            in the local timezone.
     """
 
     name: Optional[str] = None
     cron_expression: Optional[str] = None
-    start_time: Optional[datetime.datetime] = None
-    end_time: Optional[datetime.datetime] = None
-    interval_second: Optional[datetime.timedelta] = None
+    start_time: Optional[datetime] = None
+    end_time: Optional[datetime] = None
+    interval_second: Optional[timedelta] = None
     catchup: bool = False
-    run_once_start_time: Optional[datetime.datetime] = None
+    run_once_start_time: Optional[datetime] = None
+
+    @field_validator(
+        "start_time", "end_time", "run_once_start_time", mode="after"
+    )
+    @classmethod
+    def _ensure_timezone(
+        cls, value: Optional[datetime], info: ValidationInfo
+    ) -> Optional[datetime]:
+        """Ensures that all datetimes are timezone aware.
+
+        Args:
+            value: The datetime.
+            info: The validation info.
+
+        Returns:
+            A timezone aware datetime or None.
+        """
+        if value and value.tzinfo is None:
+            assert info.field_name
+            logger.warning(
+                "Your schedule `%s` is missing a timezone. It will be treated "
+                "as a datetime in your local timezone."
+            )
```

Comment:
```suggestion
            logger.warning(
                f"Your schedule `{info.field_name}` is missing a timezone. It will be treated "
                "as a datetime in your local timezone."
            )
```

Reply:
We're sort of following this style guide for our code base, but even more so, the argument of filtering log messages makes a lot of sense I think. Right now we're just logging to stdout, but we've had multiple discussions already and I'm sure at some point we will allow customizing some logging backends where ZenML logs to. So we should just write our logs with these placeholders, to be prepared for when (not if) this is being implemented.

Reply:
Yep that I fixed, thanks 🙏

Reply:
"Your schedule `%s` is missing a timezone" comment is missing a schedule :smile:

Reply:
![image](https://github.com/user-attachments/assets/2a4b63fd-ad25-4407-8a9a-98761bc3c8a5)